src/Engine/Scene/SkinningComponent.cpp

```diff
@@ -0,0 +1,505 @@
+        m_frameData.m_currentBitangent = m_refData.m_referenceMesh.getAttrib( bH ).data();
+        m_frameData.m_frameCounter     = 0;
+        m_frameData.m_doSkinning       = true;
+        m_frameData.m_doReset          = false;
+
+        // setup comp data
+        m_isReady     = true;
+        m_forceUpdate = true;
+
+        // prepare RO for skinning weights display
+        m_baseMaterial = ro->getMaterial();
+
+        // prepare UV
+        auto attrUV = Data::Mesh::getAttribName( Data::Mesh::VERTEX_TEXCOORD );
+        AttribArrayGeometry* geom;
+        if ( !m_meshIsPoly ) { geom = const_cast<TriangleMesh*>( m_triMeshWriter() ); }
+        else
+        { geom = const_cast<PolyMesh*>( m_polyMeshWriter() ); }
+        if ( geom->hasAttrib( attrUV ) )
+        {
+            auto handle = geom->getAttribHandle<Vector3>( attrUV );
+            m_baseUV    = geom->getAttrib( handle ).data();
+        }
+
+        // prepare weights material
+        auto mat  = new Data::BlinnPhongMaterial( "SkinningWeights_Mat" );
+        mat->m_kd = Color::Skin();
+        mat->m_ks = Color::White();
+
+        // assign texture
+        Data::TextureParameters texParam;
+        texParam.name = ":/Textures/Influence0.png";
+        auto tex = RadiumEngine::getInstance()->getTextureManager()->getOrLoadTexture( texParam );
+        mat->addTexture( Data::BlinnPhongMaterial::TextureSemantic::TEX_DIFFUSE, tex );
+        m_weightMaterial.reset( mat );
+
+        // compute default weights uv
+        showWeightsType( STANDARD );
+    }
+}
+
+void SkinningComponent::skin() {
+    CORE_ASSERT( m_isReady, "Skinning is not setup" );
+
+    const Skeleton* skel = m_skeletonGetter();
+
+    bool reset = ComponentMessenger::getInstance()->get<bool>( getEntity(), m_skelName );
+
+    // Reset the skin if it wasn't done before
+    if ( reset && !m_frameData.m_doReset )
+    {
+        m_frameData.m_doReset      = true;
+        m_frameData.m_frameCounter = 0;
+        m_forceUpdate              = true;
+    }
+    const auto prevPose    = m_frameData.m_skeleton.getPose( SpaceType::MODEL );
+    m_frameData.m_skeleton = *skel;
+    auto currentPose       = m_frameData.m_skeleton.getPose( SpaceType::MODEL );
+    if ( m_smartStretch ) { applySmartStretch( currentPose ); }
+    m_frameData.m_skeleton.setPose( currentPose, SpaceType::MODEL );
+    if ( !areEqual( currentPose, prevPose ) || m_forceUpdate )
+    {
+        m_forceUpdate            = false;
+        m_frameData.m_doSkinning = true;
+        m_frameData.m_frameCounter++;
+
+        const auto tH = m_refData.m_referenceMesh.getAttribHandle<Vector3>( tangentName );
+        const Vector3Array& tangents = m_refData.m_referenceMesh.getAttrib( tH ).data();
+        const auto bH = m_refData.m_referenceMesh.getAttribHandle<Vector3>( bitangentName );
+        const Vector3Array& bitangents = m_refData.m_referenceMesh.getAttrib( bH ).data();
```

Comment:
Is there a nice way to reach the tangent and bitangent vectors while in `Core` (i.e. without needing `Engine::Data::Mesh::getAttribName( VERTEX_TANGENT )`?
This would simplify all skinning functions' parameters set, and might allow to move tangent and bitangent vectors computation into `Core`.

Reply:
all attributes are managed the same way, except, position that is mandatory, and normal that is almost mandatory (and even tis one might be manage as others). So, we will always need the name to access an attribute array.

Reply:
Yes I understand that, but why isn't the name for some "standard" geometric attributes defined in `Core`, allowing the `Core` lib to create/manipulate them directly? why is there only the normal available?

Reply:
To complete, each attributes are managed the same way on the Core side,
there is a reminiscence of old fashioned way to handle attributes for normals.
Then we need to define some semantics with a naming convention, this semantic depends on the usage you want to do, and can even use "translators" from core names to engine names.

I haven't read everything in the PR, but we can imagine moving the "standard names" from Engine to Core